integration-test/ipfs.test-integration.js

```diff
@@ -0,0 +1,75 @@
+import IPFS from '../src/lib/IPFS/ipfs';
+import { config as loadEnv } from 'dotenv';
+import path from 'path';
+const file = path.join(__dirname, 'ipfs-test/infura.png');
+const folder = 'integration-test/ipfs-test';
+const unexistingFile = path.join(__dirname, 'infura-fake.png');
+
+loadEnv();
+describe('E2E Test: IPFS Service', () => {
+  jest.setTimeout(120 * 1000);
+  let ipfs;
+  let uploadedFileHash;
+  let uploadedDirectoryHash;
+
+  beforeAll(async () => {
+    ipfs = new IPFS({
+      projectId: process.env.INFURA_IPFS_PROJECT_ID,
+      projectSecret: process.env.INFURA_IPFS_PROJECT_SECRET,
+      ipfsUrl: process.env.INFURA_IPFS_ENDPOINT,
+    });
+  });
+
+  it('should upload image from local file', async () => {
+    const hash = await ipfs.uploadFile({
+      source: file,
+    });
+    uploadedFileHash = hash;
+
+    expect(hash).not.toBe(null);
+  });
+
+  it('should upload a directory', async () => {
+    const hash = await ipfs.uploadDirectory({
+      source: folder,
+    });
+
+    uploadedDirectoryHash = hash;
+
+    expect(hash).not.toBe(null);
```

Comment:
The file CID will be always the same. I think we can also check the result of the uploadDirectory to ensure we have the expected array of hashes.